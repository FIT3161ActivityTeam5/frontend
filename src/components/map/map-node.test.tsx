import React from 'react';
import renderer from 'react-test-renderer';

import MapNode from './map-node';

describe('<MapNode />', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
          <MapNode
            x={0}
            y={0}
            value={1.0}
            onDrag={jest.fn()}
            onPress={jest.fn()}
          />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('applies selected style correctly', () => {
      const tree = renderer.create(
          <MapNode
            x={0}
            y={0}
            value={1.0}
            onDrag={jest.fn()}
            onPress={jest.fn()}
            selected={true}
          />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
