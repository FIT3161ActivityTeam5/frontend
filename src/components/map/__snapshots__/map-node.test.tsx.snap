// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapNode /> applies selected style correctly 1`] = `
<RNSVGGroup
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  onMoveShouldSetResponder={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  responsible={true}
>
  <RNSVGCircle
    cx={0}
    cy={0}
    fill={4287323382}
    fillOpacity={0.5}
    propList={
      Array [
        "fill",
        "fillOpacity",
      ]
    }
    r={15}
  />
  <RNSVGCircle
    cx={0}
    cy={0}
    fill={4287323382}
    propList={
      Array [
        "fill",
      ]
    }
    r={12}
  />
  <RNSVGCircle
    cx={0}
    cy={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    r={18}
    stroke={4286331629}
    strokeWidth={3}
  />
  <RNSVGText
    content={null}
    dx={Array []}
    dy={Array []}
    fill={4294967295}
    font={
      Object {
        "textAnchor": "middle",
      }
    }
    propList={
      Array [
        "fill",
      ]
    }
    rotate={Array []}
    x={
      Array [
        0,
      ]
    }
    y={
      Array [
        4,
      ]
    }
  >
    <RNSVGTSpan
      content="1.0"
      dx={Array []}
      dy={Array []}
      font={Object {}}
      rotate={Array []}
      x={Array []}
      y={Array []}
    />
  </RNSVGText>
</RNSVGGroup>
`;

exports[`<MapNode /> renders correctly 1`] = `
<RNSVGGroup
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  onMoveShouldSetResponder={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  responsible={true}
>
  <RNSVGCircle
    cx={0}
    cy={0}
    fill={4287323382}
    fillOpacity={0.5}
    propList={
      Array [
        "fill",
        "fillOpacity",
      ]
    }
    r={15}
  />
  <RNSVGCircle
    cx={0}
    cy={0}
    fill={4287323382}
    propList={
      Array [
        "fill",
      ]
    }
    r={12}
  />
  <RNSVGText
    content={null}
    dx={Array []}
    dy={Array []}
    fill={4294967295}
    font={
      Object {
        "textAnchor": "middle",
      }
    }
    propList={
      Array [
        "fill",
      ]
    }
    rotate={Array []}
    x={
      Array [
        0,
      ]
    }
    y={
      Array [
        4,
      ]
    }
  >
    <RNSVGTSpan
      content="1.0"
      dx={Array []}
      dy={Array []}
      font={Object {}}
      rotate={Array []}
      x={Array []}
      y={Array []}
    />
  </RNSVGText>
</RNSVGGroup>
`;
